provider/pkg/provider/provider.go

```diff
@@ -1034,12 +1034,23 @@ func (k *kubeProvider) Check(ctx context.Context, req *pulumirpc.CheckRequest) (
 
 	var failures []*pulumirpc.CheckFailure
 
-	// If annotations with a reserved internal prefix exist, ignore them.
+	hasHelmHook := false
 	for key := range newInputs.GetAnnotations() {
+		// If annotations with a reserved internal prefix exist, ignore them.
 		if metadata.IsInternalAnnotation(key) {
 			_ = k.host.Log(ctx, diag.Warning, urn,
 				fmt.Sprintf("ignoring user-specified value for internal annotation %q", key))
 		}
+
+		// If the Helm hook annotation is found, set the hasHelmHook flag.
+		if has := metadata.IsHelmHookAnnotation(key); has {
+			hasHelmHook = hasHelmHook || has
+		}
+	}
+	if hasHelmHook {
+		_ = k.host.Log(ctx, diag.Warning, urn,
+			"this resource contains helm hooks that are not currently supported. " +
+			"This feature is tracked at https://github.com/pulumi/pulumi-kubernetes/issues/555")
```

Comment:
Nit - Instead of specifying the bug (since its quite busy and has a lot on it), might be better to update the verbiage to specify the impact of this to the user? i.e. something like `this resource contains helm hooks that are currently not supported. You may have to perform the hook related operation (%s) manually.` where `%s` is the annotation value?